### PR TITLE
gh-118986: expose socket.IPV6_RECVERR

### PR DESCRIPTION
**Pull request by chrysn**
This adds two socket option numbers, without which the existing `MSG_ERRQUEUE` gives barely any benefits.

* Issue: gh-118986

### CODE REVIEW

**Comment by chrysn** [comment]
This has missed the 3.13 alpha; may I ask for a review again to have this available next round?

@Eclips4 and @jeremyhylton would be suitable reviewers given the history of https://github.com/python/cpython/issues/120056.

File: Misc/NEWS.d/next/Library/2024-05-13-10-09-41.gh-issue-118986.-r4W9h.rst
Comment (Eclips4) [inline Misc/NEWS.d/next/Library/2024-05-13-10-09-41.gh-issue-118986.-r4W9h.rst]:
```suggestion
Add :data:`!socket.IPV6_RECVERR` constant (available since Linux 2.2).
```

**Comment by Eclips4** [comment]
Please, modify the lines stated below to include the `IPV6_RECVERR`.
https://github.com/python/cpython/blob/0d88b995a641315306d56fba7d07479b2c5f57ef/Doc/library/socket.rst?plain=1#L453-L455

**Comment by chrysn** [comment]
Adjusted as requested.

You pointed at the lines for Python 3.14 -- do I read this right that this PR can become part of 3.14 still?

**Comment by Eclips4** [comment]
> You pointed at the lines for Python 3.14 -- do I read this right that this PR can become part of 3.14 still?

Yes, you understand that correctly! Python 3.14 is in development stage and two days ago was released the first alpha of 3.14, so we have more than half a year to implement new features for 3.14 version. For more information see https://peps.python.org/pep-0745/

**Review by Eclips4 — APPROVED** [review]
Thanks! I'll let it unmerged for some time to get a review from others if they have something to say.

File: Modules/socketmodule.c
Comment (sobolevn) [inline Modules/socketmodule.c]:
```suggestion
#ifdef IPV6_RECVERR
```

style nit

**Review by sobolevn — APPROVED** [review]
Thanks!

File: Modules/socketmodule.c
Comment (chrysn) [reply Modules/socketmodule.c]: How did that happen? Anyway, thanks, fixed.

**Comment by Eclips4** [comment]
@chrysn, in the future, please avoid force pushes as they make PRs difficult to review. Commits are squashed anyway before merging, so it doesn't matter how many commits you need to complete a PR :)

**Comment by Eclips4** [comment]
> This adds two socket option numbers, without which the existing `MSG_ERRQUEUE` gives barely any benefits.
> 
>     * Issue: [socket.IP{,V6}_RECVERR not exposed #118986](https://github.com/python/cpython/issues/118986)

You've said that there are two new socket constants, but in fact, this PR contains only one. Did you forget to add the second one?

**Comment by chrysn** [comment]
The issue was originally about adding two constants (I've updated the title but kept the original post).

One of the constants was already added after this PR was filed in a different PR (https://github.com/python/cpython/issues/120056), so when I updated this PR, only one was left.

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thank you!

**Pull request merged**
